### PR TITLE
Change /api/v1/items/aggregations filtering logic

### PR DESCRIPTION
**Pull request by eronisko**
* added tests
* facet now does not affect itself (but just other facets)
* filtered (i.e. selected) value are always included in the aggregation results, so that they don't disappear

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)

# Checklist:

- [x] I have updated the [CHANGELOG](../CHANGELOG.md)
- [x] I have requested at least 1 reviewer for this PR
- [ ] I have made corresponding changes to the documentation

### CODE REVIEW

File: app/Http/Controllers/Api/V1/ItemController.php [hunk 1]
```diff
@@ -101,55 +100,76 @@ public function aggregations(Request $request)
...
-                    'size' => $size,
-                ],
-            ]);
+            $aggregationsQuery[$agg]['aggregations']['filtered']['terms'] = [
+                'field' => $field,
+                'size' => $size,
+            ];
         }
 
         foreach ($min as $agg => $field) {
```
Comment (FrantisekMichalSebestyen) [inline app/Http/Controllers/Api/V1/ItemController.php:112]: I would mabye rename this. Usually you would want the elements to be named more concrete than array.
Analogical example:
```
**Bad**
foreach($tigers as $animal)
**Good**
foreach($tigers as $tiger)
```

File: app/Http/Controllers/Api/V1/ItemController.php [hunk 2]
```diff
@@ -101,55 +100,76 @@ public function aggregations(Request $request)
...
+            )->buildQuery();
+        }
+
+        $searchRequest = Item::searchQuery()
+            ->size(0)
+            ->aggregateRaw([
+                'all_items' => [
+                    'global' => (object) [],
+                    'aggregations' => (object) $aggregationsQuery,
                 ],
```
Comment (FrantisekMichalSebestyen) [inline app/Http/Controllers/Api/V1/ItemController.php:139]: This is unclear to me.
Why `->size(0)`?

File: app/Http/Controllers/Api/V1/ItemController.php [hunk 3]
```diff
@@ -101,55 +100,76 @@ public function aggregations(Request $request)
...
         foreach ($min as $agg => $field) {
-            $searchRequest->aggregate($agg, [
-                'min' => [
-                    'field' => $field,
-                ],
-            ]);
+            $aggregationsQuery[$agg]['aggregations']['filtered']['min'] = [
+                'field' => $field,
+            ];
         }
```
Comment (FrantisekMichalSebestyen) [inline app/Http/Controllers/Api/V1/ItemController.php:116]: I would mabye rename this. Usually you would want the elements to be named more concrete than array.
Analogical example:
```
**Bad**
foreach($tigers as $animal)
**Good**
foreach($tigers as $tiger)
```

File: app/Http/Controllers/Api/V1/ItemController.php [hunk 4]
```diff
@@ -101,55 +100,76 @@ public function aggregations(Request $request)
...
+                );
+            });
+
+        // Ensure filtered (i.e. selected) terms are included in the response
+        foreach ($filter as $term => $value) {
+            if (!Arr::has($terms, $term)) {
+                continue;
+            }
+
+            foreach (array_reverse(Arr::wrap($value)) as $value) {
```
Comment (FrantisekMichalSebestyen) [inline app/Http/Controllers/Api/V1/ItemController.php:165]: `array_reverse` ?

**Review by FrantisekMichalSebestyen — APPROVED** [review]

**Comment by eronisko** [comment]
@FrantisekMichalSebestyen I've added the exception for `date_earliest`/`date_latest`

**Pull request merged**